Polyhedron/demo/Polyhedron/Plugins/Mesh_3/Mesh_3_plugin.cpp

```diff
@@ -709,13 +709,17 @@ void Mesh_3_plugin::mesh_3(const Mesh_type mesh_type,
       return;
     }
 #ifdef CGAL_USE_ITK
-    if (sigma_weights > 0 && nullptr == image_item->image_weights())
+    if ( sigma_weights > 0
+      && sigma_weights != image_item->sigma_weights())
```

Comment:
You should deal with the case where a first run was with a positive sigma, and then the user reset sigma to 0.f.

Reply:
if the user resets sigma to 0, we do not want to compute the weights (it will crash), and `pWeights` below will be set to `nullptr`, so the weights are not used.
Isn't this was we want?

Reply:
Yes, indeed. I did not know `pWeights` was modified as well.